nrf52840-hal/src/ieee802154.rs

```diff
@@ -243,6 +259,50 @@ impl<'c> Radio<'c> {
             .write(|w| w.txpower().variant(power._into()));
     }
 
+    /// Sample the received signal power (i.e. the presence of possibly interfering signals)
+    /// within the bandwidth of the currently used channel for sample_cycles iterations.
+    /// The result can be used to find a suitable ED threshold for Energy Detection-based CCA mechanisms.
+    ///
+    /// Returns the *maximum* measurement recorded during sampling as reported by the hardware (not in dBm!)
+    ///
+    /// For details, see Section 6.20.12.3 Energy detection (ED) of the PS.
+    /// RSSI samples are averaged over a measurement time of 8 symbol periods (128 μs).
+    pub fn energy_detection_scan(&mut self, sample_cycles: u32) -> u8 {
+        self.needs_enable = true;
+
+        // clear related events
+        self.radio.events_edend.reset();
+        // ensure that the shortcut between READY event and START task is disabled before putting
+        // the radio into recv mode
+        self.radio.shorts.reset();
+        self.put_in_rx_mode();
+
+        dma_start_fence();
```

Comment:
this and the other fence (`dma_end_fence`) are not required because we are not using the DMA engine in this function; the two fences should be removed

Reply:
gah, leftovers from my debugging spree. fixed.